projects/habitat_ovmm/memory_data_collection.py

```diff
@@ -0,0 +1,108 @@
+import argparse
+import json
+import os
+import sys
+import time
+from collections import defaultdict
+from pathlib import Path
+
+import numpy as np
+from config_utils import get_config
+from eval_vectorized import VectorizedEvaluator
+from omegaconf import DictConfig, OmegaConf
+
+sys.path.insert(
+    0,
+    str(Path(__file__).resolve().parent.parent.parent / "src/home_robot"),
+)
+sys.path.insert(
+    0,
+    str(Path(__file__).resolve().parent.parent.parent / "src/home_robot_sim"),
+)
+
+from habitat import make_dataset
+from habitat.core.environments import get_env_class
+from habitat.core.vector_env import VectorEnv
+from habitat.utils.gym_definitions import _get_env_name
+from habitat_baselines.rl.ppo.ppo_trainer import PPOTrainer
+
+from home_robot.agent.ovmm_agent.llm_agent import LLMAgent
+from home_robot.agent.ovmm_agent.ovmm_agent import OpenVocabManipAgent
+from home_robot_sim.env.habitat_ovmm_env.habitat_ovmm_env import (
+    HabitatOpenVocabManipEnv,
+)
+
+
+def create_ovmm_env_fn(config):
+    """Create habitat environment using configsand wrap HabitatOpenVocabManipEnv around it. This function is used by VectorEnv for creating the individual environments"""
+    habitat_config = config.habitat
+    dataset = make_dataset(habitat_config.dataset.type, config=habitat_config.dataset)
+    env_class_name = _get_env_name(config)
+    env_class = get_env_class(env_class_name)
+    habitat_env = env_class(config=habitat_config, dataset=dataset)
+    habitat_env.seed(habitat_config.seed)
+    env = HabitatOpenVocabManipEnv(habitat_env, config, dataset=dataset)
+    return env
+
+
+class MemoryCollector(VectorizedEvaluator):
+    """Class for creating vectorized environments, evaluating OpenVocabManipAgent on an episode dataset and returning metrics"""
+
+    def __init__(self, config, config_str: str):
+        super().__init__(config, config_str)
+
+    def run(self, num_episodes_per_env=10):
+        self._init_envs(
+            config=self.config, is_eval=True, make_env_fn=create_ovmm_env_fn
+        )
+        agent = LLMAgent(
+            config=self.config,
+            obs_spaces=self.envs.observation_spaces,
+            action_spaces=self.envs.orig_action_spaces,
+        )
+        self._eval(
+            agent,
+            self.envs,
+            num_episodes_per_env=num_episodes_per_env,
+            episode_keys=None,
+        )
+        with open("datadump/memory_data.json", "w") as json_file:
+            json.dump(agent.memory, json_file)
```

Comment:
What is this expected to return? what fields are in this file

Reply:
It will return the robot memory in the format of a dict. Just updated this fn to improve clarity